exec/java-exec/src/test/java/org/apache/drill/exec/record/TestVectorContainer.java

```diff
@@ -124,4 +132,52 @@ public void testContainerMerge() {
     leftIndirect.clear();
     right.clear();
   }
+
+  @Test
+  public void testAppendRow()
+  {
+    MaterializedField colA = MaterializedField.create("colA", Types.required(TypeProtos.MinorType.INT));
+    MaterializedField colB = MaterializedField.create("colB", Types.required(TypeProtos.MinorType.INT));
```

Comment:
Maybe add some "interesting" datatypes ?  Testing integers only may miss some issue.

Reply:
Added more interesting types. Currently RowSet classes don't support the Map data type. Paul asked me to look into adding support for this a while ago for DRILL-5870 . I'll update the test framework to support that in the next PR.